File: android/src/main/java/com/mycompany/plugins/example/PaysafePlugin.java
package com.mycompany.plugins.example;

import android.content.Context;
import android.util.Log;

import androidx.activity.ComponentActivity;

import com.paysafe.android.core.domain.exception.PaysafeException;
import com.paysafe.android.core.domain.exception.PaysafeRuntimeError;


public class PaysafePlugin {
    private final PaysafeSDKService paysafeSDKService = new PaysafeSDKService();

    public void setupSDK() {
        paysafeSDKService.setupSDK();
    }

    public void initializeVenmo(ComponentActivity activity, VenmoInitializationCallback completion) {
        paysafeSDKService.initializeVenmo(activity, completion);
    }

    // Method to tokenize with a consumerId and a completion callback
    public void tokenize(Context context, String consumerId, VenmoTokenizationCallback completion) {
        paysafeSDKService.tokenize(context, consumerId, completion);
    }
}

File: demo-plugin-kotlin/android/src/main/java/com/mycompany/plugins/example/PaysafePlugin.kt
package com.mycompany.plugins.example

import android.content.Context
import androidx.activity.ComponentActivity
import kotlin.coroutines.Continuation

class PaysafePlugin {
    private val paysafeSDKService = PaysafeSDKService()

    fun setupSDK() {
        paysafeSDKService.setupSDK()
    }

    fun initializeVenmo(activity: ComponentActivity, completion: VenmoInitializationCallback) {
        paysafeSDKService.initializeVenmo(activity) {
            completion.onInitialized()
        }
    }

    // Method to tokenize with a consumerId and a completion callback
    suspend fun tokenize(context: Context?, consumerId: String?, completion: VenmoTokenizationCallback) {
        paysafeSDKService.tokenize(context, consumerId) {
            completion.onTokenized(it)
        }
    }
}

